starknet_py/net/client_models.py

```diff
@@ -284,10 +284,17 @@ class ContractsNonce:
     nonce: int
 
 
+@dataclass
+class DeclaredContractHash:
+    class_hash: int
+    compiled_class_hash: int
+
+
 @dataclass
 class StateDiff:
     deployed_contracts: List[DeployedContract]
-    declared_contract_hashes: List[int]
+    deprecated_declared_contract_hashes: List[int]
+    declared_contract_hashes: List[DeclaredContractHash]
     storage_diffs: List[StorageDiffItem]
     nonces: List[ContractsNonce]
```

Comment:
Gateway's state diff now includes a `replaced_classes` section that is not present in the RPC master branch. Should it be added as an optional field or omitted?

Reply:
I think we shouldn't care for the new RPC compliance right now. It will be taken care of, once a new version comes out.

Reply:
Turns out it will be added to the RPC, will add it to the dataclass as well.

https://github.com/starkware-libs/starknet-specs/pull/76

Reply:
It's a list, so we don't have to make it optional - we can default to an empty list.